lib/benchpark/experiment.py

```diff
@@ -129,10 +129,18 @@ class Experiment(ExperimentSystemBase, SingleNode):
 
     def __init__(self, spec):
         self.spec: "benchpark.spec.ConcreteExperimentSpec" = spec
+        # Device type must be set before super with absence of mpionly experiment type
+        self.device_type = "cpu"
```

Comment:
@pearce8 MPI-only experiments will be set here, because the will still inherit from the `Experiment` class.